hepcrawl/spiders/aps_spider.py

```diff
@@ -58,64 +65,312 @@ def __init__(self, url=None, from_date=None, until_date=None,
-
-            # record.add_xpath('license', '//license/license-p/ext-link/text()')
-            # record.add_xpath('license_type', '//license/@license-type')
-            record.add_value('license_url', get_nested(article, 'rights', 'licenses')[0]['url'])
-
-            record.add_value('collections', ['HEP', 'Citeable', 'Published'])
-            yield record.load_item()
+            doi = article["identifiers"]["doi"]
+            link = "http://harvest.aps.org/v2/journals/articles/" + doi
+            # FIXME: sometimes requests are forbidden? Why??
+            # TEST: local test file:
+            # link = "file:///home/henrikv/.virtualenvs/hepcrawl/src/hepcrawl/tests/responses/aps/aps_single_response_test_ref.xml"
+
+            request = Request(
+                link, headers={'Accept': 'text/xml'}, callback=self.scrape_xml_for_ref)
+            request.meta["article"] = article
+            yield request
 
         # Pagination support. Will yield until no more "next" pages are found
         if 'Link' in response.headers:
             links = link_header.parse(response.headers['Link'])
-            next = links.links_by_attr_pairs([('rel', 'next')])
-            if next:
-                next_url = next[0].href
+            next_link = links.links_by_attr_pairs([('rel', 'next')])
+            if next_link:
+                next_url = next_link[0].href
                 yield Request(next_url)
 
+    @staticmethod
+    def _get_journal_and_section(publication):
+        """Take journal title data (with possible section) and try to fix it."""
+        # FIXME: this is also in Elsevier harvest, move to utils?
+        section = ''
+        journal_title = ''
+        possible_sections = ["A", "B", "C", "D", "E"]
+        try:
+            # filter after re.split, which may return empty elements:
+            split_pub = filter(None, re.split(r'(\W+)', publication))
+            if split_pub[-1] in possible_sections:
+                section = split_pub.pop(-1)
+            journal_title = "".join(
+                [word for word in split_pub if "section" not in word.lower()]).strip(", ")
+        except IndexError:
+            pass
+
+        return journal_title, section
+
+    @staticmethod
+    def _format_arxiv_id(arxiv_urls):
+        """Return an arxiv id with format 'arxiv:1407.0275'.
+
+        For old identifiers format is 'hep-ex/9908047'. Input string may or
+        may not have an 'http://' in it.
+        """
+        # FIXME: this is also in Elsevier harvest, move to utils?
+        # Or do we want this at all?
+        if arxiv_urls:
+            arxiv_id = arxiv_urls[0].split(":")[-1]
+            if arxiv_id and '.' not in arxiv_id:
+                return arxiv_id.strip("/")
+            else:
+                return u"arxiv:{}".format(arxiv_id)
+
+    @staticmethod
+    def _clean_text(text):
+        """Replace all occurences of certain words in a string."""
```

Comment:
Trying to format the raw reference nicely. The original raw string is a mess because the XML has mixed content. One of my questions is related to this: do we want the raw reference to be a text or xml string? If we want xml string, this formatting can be removed.